dialogue-apache-hc4-client/src/main/java/com/palantir/dialogue/hc4/ApacheHttpClientChannels.java

```diff
@@ -96,6 +100,26 @@ public static CloseableClient createCloseableHttpClient(ClientConfiguration conf
         long socketTimeoutMillis =
                 Math.max(conf.readTimeout().toMillis(), conf.writeTimeout().toMillis());
         int connectTimeout = Ints.checkedCast(conf.connectTimeout().toMillis());
+
+        SocketConfig socketConfig = SocketConfig.custom().setSoKeepAlive(true).build();
+        SSLConnectionSocketFactory sslSocketFactory = new SSLConnectionSocketFactory(
+                conf.sslSocketFactory(),
+                new String[] {"TLSv1.2"},
+                conf.enableGcmCipherSuites()
+                        ? jvmSupportedCipherSuites(CipherSuites.allCipherSuites())
+                        : jvmSupportedCipherSuites(CipherSuites.fastCipherSuites()),
+                new DefaultHostnameVerifier());
+
+        PoolingHttpClientConnectionManager connectionManager = new PoolingHttpClientConnectionManager(
+                RegistryBuilder.<ConnectionSocketFactory>create()
+                        .register("http", PlainConnectionSocketFactory.getSocketFactory())
+                        .register("https", sslSocketFactory)
```

Comment:
I wonder if we can wrap these to add a meter  of connection rate, and wrap the pool manager to capture the rate connections requested?

Reply:
Mmm that would be cool, so dividing one by the other tells us how effectively we're re-using... Do we not already get that connection rate from the tritium tls.handshakes?

Reply:
Tritium handshake instrumentation would work assuming https, but I don’t think we wire it up yet.